src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -3008,9 +3009,27 @@ Node TheoryStringsRewriter::rewriteStrConvert(Node node)
       newChar = CVC4::String::convertCodeToUnsignedInt(newChar);
       nvec[i] = newChar;
     }
-    Node retNode = NodeManager::currentNM()->mkConst(String(nvec));
+    Node retNode = nm->mkConst(String(nvec));
     return returnRewrite(node, retNode, "str-conv-const");
   }
+  else if (node[0].getKind() == STRING_CONCAT)
+  {
+    std::vector<Node> cc;
+    for (const Node& nc : node[0])
+    {
+      cc.push_back(nm->mkNode(nk, nc));
+    }
+    Node retNode = nm->mkNode(STRING_CONCAT, cc);
+    return returnRewrite(node, retNode, "str-conv-minscope-concat");
+  }
+  else if (node[0].getKind() == STRING_TOLOWER
+           || node[0].getKind() == STRING_TOUPPER)
+  {
+    // tolower( tolower( x ) ) = tolower( x )
+    // tolower( toupper( x ) ) = tolower( x )
+    Node retNode = nm->mkNode(nk, node[0][0]);
+    return returnRewrite(node, retNode, "str-conv-idem");
```

Comment:
Another basic rewrite would be: `tolower/toupper(int.to.str(n)) ---> int.to.str(n)` (maybe future PR).

Reply:
Ah, yeah, thats a good point, I'll add.